packages/fiori/src/Timeline.ts

```diff
@@ -105,51 +117,103 @@ class Timeline extends UI5Element {
 
+		if (this.items) {
+			for (let i = 0; i < this.items.length; i++) {
+				this.items[i].layout = this.layout;
+				if (this.items[i + 1] && !!this.items[i + 1].icon) {
+					this.items[i].forcedLineWidth = SHORT_LINE_WIDTH;
+				} else if (this.items[i].icon && this.items[i + 1] && !this.items[i + 1].icon) {
+					this.items[i].forcedLineWidth = LARGE_LINE_WIDTH;
+				}
+			}
+		}
+		this._setLastItem();
+		this._setIsNextItemGroup();
+		this.items[0]._firstItemInTimeline = true;
+	}
+
+	_setLastItem() {
+		const items = this.items;
+		if (items && items.length > 0) {
+			items[items.length - 1]._lastItem = true;
+		}
+	}
+
+	_setIsNextItemGroup() {
 		for (let i = 0; i < this.items.length; i++) {
-			this.items[i].layout = this.layout;
-			if (this.items[i + 1] && !!this.items[i + 1].icon) {
-				this.items[i].forcedLineWidth = SHORT_LINE_WIDTH;
-			} else if (this.items[i].icon && this.items[i + 1] && !this.items[i + 1].icon) {
-				this.items[i].forcedLineWidth = LARGE_LINE_WIDTH;
+			if (this.items[i + 1] && this.items[i + 1].isGroupItem) {
+				this.items[i]._isNextItemGroup = true;
 			}
 		}
 	}
 
 	_onkeydown(e: KeyboardEvent) {
-		const target = e.target as TimelineItem;
+		const target = e.target as ITimelineItem;
 
 		if (isTabNext(e)) {
 			if (!target.nameClickable || getEventMark(e) === "link") {
-				this._handleTabNextOrPrevious(e, isTabNext(e));
+				this._handleNextOrPreviousItem(e, true);
 			}
 		} else if (isTabPrevious(e)) {
-			this._handleTabNextOrPrevious(e);
+			if ((!target.nameClickable || getEventMark(e) === "link")) {
+				this._handleNextOrPreviousItem(e);
+			}
 		}
 	}
 
-	_handleTabNextOrPrevious(e: KeyboardEvent, isNext?: boolean) {
-		const target = e.target as TimelineItem;
-		const nextTargetIndex = isNext ? this.items.indexOf(target) + 1 : this.items.indexOf(target) - 1;
-		const nextTarget = this.items[nextTargetIndex] as TimelineItem;
+	_handleNextOrPreviousItem(e: KeyboardEvent, isNext?: boolean) {
+		const target = e.target as ITimelineItem | ToggleButton;
+		let updatedTarget = target;
+
+		if ((target as ITimelineItem).isGroupItem) {
+			updatedTarget = target.shadowRoot!.querySelector<ToggleButton>("ui5-toggle-button")!;
+		}
+
+		const nextTargetIndex = isNext ? this._navigatableItems.indexOf(updatedTarget) + 1 : this._navigatableItems.indexOf(updatedTarget) - 1;
+		const nextTarget = this._navigatableItems[nextTargetIndex] as ITimelineItem | TimelineGroupItem | ToggleButton;
+
 		if (!nextTarget) {
 			return;
 		}
-		if (nextTarget.nameClickable && !isNext) {
+
+		if (nextTarget) {
 			e.preventDefault();
-			nextTarget.focusLink();
-			return;
+			nextTarget.focus();
+			this._itemNavigation.setCurrentItem(nextTarget);
 		}
-		e.preventDefault();
-		nextTarget.focus();
-		this._itemNavigation.setCurrentItem(nextTarget);
+	}
+
+	get _navigatableItems() {
+		const navigatableItems: Array<ITimelineItem | ToggleButton> = [];
+		this.items.forEach(item => {
```

Comment:
Write early return to ensure that we have items. We can return empty array, to ensure that ItemNavigation gets valid data.